modules/promotion/src/Plugin/Field/FieldWidget/UsageLimitPerClientWidget.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Drupal\commerce_promotion\Plugin\Field\FieldWidget;
+
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\WidgetBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Plugin implementation of 'commerce_usage_limit_per_client'.
+ *
+ * @FieldWidget(
+ *   id = "commerce_usage_limit_per_client",
+ *   label = @Translation("Usage limit per client"),
+ *   field_types = {
+ *     "integer"
+ *   }
+ * )
+ */
+class UsageLimitPerClientWidget extends WidgetBase {
+
+  protected $parent;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state) {
+    $value = isset($items[$delta]->value) ? $items[$delta]->value : NULL;
```

Comment:
Explain why you get data from parent and what does radio_path is used for

Reply:
Comments added. 
I tried to be as clear as possible but do not hesitate to modify if necessary.